bundler/spec/dependabot/bundler/update_checker/version_resolver_spec.rb

```diff
@@ -47,68 +46,42 @@
   end
   let(:source) { nil }
   let(:requirement_string) { ">= 0" }
-
-  let(:gemfile) do
-    Dependabot::DependencyFile.new(
-      content: fixture("ruby", "gemfiles", gemfile_fixture_name),
-      name: "Gemfile"
-    )
-  end
-  let(:lockfile) do
-    Dependabot::DependencyFile.new(
-      content: fixture("ruby", "lockfiles", lockfile_fixture_name),
-      name: "Gemfile.lock"
-    )
-  end
-  let(:gemspec) do
-    Dependabot::DependencyFile.new(
-      content: fixture("ruby", "gemspecs", gemspec_fixture_name),
-      name: "example.gemspec"
-    )
-  end
-  let(:gemfile_fixture_name) { "Gemfile" }
-  let(:lockfile_fixture_name) { "Gemfile.lock" }
-  let(:gemspec_fixture_name) { "example" }
   let(:rubygems_url) { "https://index.rubygems.org/api/v1/" }
 
   describe "#latest_resolvable_version_details" do
     subject { resolver.latest_resolvable_version_details }
 
     context "with a rubygems source" do
       context "with a ~> version specified constraining the update" do
-        let(:gemfile_fixture_name) { "Gemfile" }
-        let(:lockfile_fixture_name) { "Gemfile.lock" }
         let(:requirement_string) { "~> 1.4.0" }
 
+        let(:dependency_files) { project_dependency_files("bundler1/Gemfile_Gemfile") }
```

Comment:
Would it work if we uniqued these names so this ends up as `Gemfile`?

Reply:
Yeah, but we need to also handle the scenario where there's only a Gemfile too

Reply:
Maybe we can figure out if there's a lockfile, and if not we append `_no_lockfile`